clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -456,6 +456,20 @@ public static String formatAddress(String host, Integer port) {
         return sb.toString();
     }
 
+    public static <K, V> String join(Map<K, V> map,
```

Comment:
There is a `mkString` method in the same class so maybe we should use that name to be consistent. Or we could rename the other one to `join` (which is what the Java standard library uses).